### DISCUSSION TITLE
Generating classes from WSDL file including scheme files

### DISCUSSION THREAD

**Discussion by trailsnail** [phpro/soap-client#392] [Q&A] [answered]
Hi,

I have used your Soap Client to generate PHP classes from a WSDL, which itself also includes schema files.
I noticed something, but I don't know if it's a mistake and where it might have happened.

The scheme describes the following:
![grafik](https://user-images.githubusercontent.com/4742603/138097652-61bebc1b-7a02-44eb-9493-f51169787787.png)

I want to extend an instance of  `Shipment` with an instance of `ShipmentDetailsType` to set `accountNumber` and `product` which are properties of `ShipmentDetailsType`.

In addition, a `ShipmentItem` must be attached. However, this only exists in the class `ShipmentDetailsTypeType`.

![grafik](https://user-images.githubusercontent.com/4742603/138097371-454d5467-60d0-4899-9e40-e2e1058b869b.png)

So back to the scheme: have the classes been created incorrectly?
Because in order to create a valid XML object, I had to manually extend the generated class `ShipmentDetailsTypeType` by `extends ShipmentDetailsType` - this is my interpretation from the schema.
I changed the porperties from `private` to `protected`.

Now the XML looks the way it should look when I fill data

```
<ShipmentOrder xsi:type="ns2:ShipmentOrderType">
  <sequenceNumber/>
  <Shipment>
	<ShipmentDetails>
	  <product>V01PAK</product>
	  <ns1:accountNumber>22222222220104</ns1:accountNumber>
	  <customerReference>Ref. 123456</customerReference>
	  <shipmentDate>2021-10-20</shipmentDate>
	  <ShipmentItem>
		<weightInKG>1.3</weightInKG>
	  </ShipmentItem>
	</ShipmentDetails>
  </Shipment>
</ShipmentOrder>
```

Any ideas, opinions or experiences?

This is my first week dealing with this topic.

Thank you

**Comment by veewee** [comment, marked as answer]
Ext-soap does not provide that information:
https://bugs.php.net/bug.php?id=45404

This issue has been reported a few times: #311, #291, #156, #144 ...

The solution is by adding a manual configuration that points out which classes need to extend from which classes by using the extend assembler:
https://github.com/phpro/soap-client/blob/master/docs/code-generation/assemblers.md#extendassembler

Until we have a better way of getting the data back from the WSDL, there is no simple solution for this.

**Reply by trailsnail** [reply]
Wow. Thank you.

Since the problem has been named several times, I am glad that I only have one of these classes in the project that I have to adjust manually (currently).
